server/controllers/profileController.js

```diff
@@ -0,0 +1,149 @@
+
+import Model from '../models';
+
+const { Profile, User } = Model;
+
+/**
+  * @class ProfileController
+  * @description CRUD operations on Profile
+  */
+export default class ProfileController {
+  /**
+  * @description -This method creates profile for a authenticated user
+  * @param {object} req - The request payload sent from the router
+  * @param {object} res - The response payload sent back from the controller
+  * @returns {object} - status, message and profile detail
+  */
+  static createOrUpdateProfile(req, res) {
+    const body = ProfileController.updateReqBody(req);
+
+    return Profile.findOne({
+      where: { username: req.user.username },
+    })
+      .then((profile) => {
+        if (profile) {
+          return profile.update(body)
+            .then(data => ({ created: false, data }));
+        }
+        return Profile.create(body)
+          .then(data => ({ created: true, data }));
+      })
+      .then(({ created, data }) => {
+        if (created) {
+          return res.status(201).json({ success: true, message: 'Profile created successfully', profile: data });
+        }
+        return res.status(200).json({ success: true, message: 'Profile updated successfully', profile: data });
+      })
+      .catch(() => res.status(500).json({ success: false, error: 'Profile could not be updated' }));
+  }
+
+  /**
+   * @description -This method get profile detail by id
+   * @param {object} req - The request payload sent to the router
+   * @param {object} res - The response payload sent back from the controller
+   * @returns {object} - status, message and profile detail
+   */
+  static getProfile(req, res) {
+    return Profile.findOne({
+      where: { username: req.user.username },
+      include: [{
+        model: User,
+        attributes: ['username', 'email', 'createdAt', 'updatedAt']
+      }]
+    })
+      .then((profile) => {
+        res.status(200).json({ success: true, message: 'Profile fetched successfully', profile });
+      })
+      .catch(() => res.status(500).json({ success: false, error: 'Failed to fetch profile' }));
+  }
+
+  /**
+   * @description -This method get all profile details
+   * @param {object} req - The request payload sent from the router
+   * @param {object} res - The response payload sent back from the controller
+   * @returns {object} - status, message and profile detail
+   */
+  static getProfiles(req, res) {
```

Comment:
I think getOneProfile and getAllProfile (something explicit) would have been better for the method names.

Reply:
Done